packages/office-ui-fabric-react/src/components/Icon/examples/Icon.RegisterIconRenderer.Example.tsx

```diff
@@ -0,0 +1,70 @@
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import { Icon, IconType, IIconProps, registerIconRenderer } from 'office-ui-fabric-react/lib/Icon';
+import './IconExample.scss';
+
+export class IconRegisterIconRendererExample extends React.Component<any, any> {
+  private _unsubscribeRendererFn: () => void;
+
+  public componentWillMount() {
+    this._unsubscribeRendererFn = registerIconRenderer((iconProps: IIconProps) => {
+      // This function can return any valid JSX.Element icon (e.g. an svg, an svg use reference, an img that points to a png/bmp, a sprite w/ appropriate css styling applied, etc.)
+      if (iconProps.iconName === 'Dot') {
+        return (
+          <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100'>
+            <circle cx='50' cy='50' r='40' fill='orange' />
+          </svg>);
+      }
+      return iconProps.iconName === 'OneDrive' ? oneDriveIcon() :
```

Comment:
nested ternaries make my head hurt, can we split this out into if statements?

Reply:
even for a return statement?